src/main/webapp/app/faq/faq.component.ts

```diff
@@ -64,11 +75,21 @@ export class FaqComponent implements OnInit, OnDestroy {
         this.searchInput.pipe(debounceTime(300)).subscribe((searchTerm: string) => {
             this.refreshFaqList(searchTerm);
         });
+        this.routeDataSubscription = this.route.data.subscribe((data) => {
+            const course = data['course'];
+            if (course) {
+                this.course = course;
+                this.isAtLeastInstructor = this.accountService.isAtLeastInstructorInCourse(course);
+            }
+        });
     }
 
     ngOnDestroy(): void {
         this.dialogErrorSource.complete();
         this.searchInput.complete();
+        if (this.routeDataSubscription) {
+            this.routeDataSubscription.unsubscribe();
+        }
```

Comment:
You do not need to dismiss reviews for this little change, this would be a simpler way to express the same thing
```suggestion
         this.routeDataSubscription?.unsubscribe();
```